### PR TITLE
SDI-505 Handle activities.activity.created event

### PR DESCRIPTION
**Pull request by steverendell**
Create activity api service

### CODE REVIEW

File: docker-compose-localstack.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
       - hmpps
     container_name: localstack
     ports:
-      - "4566-4597:4566-4597"
+      - "4566:4566"
```
Comment (steverendell) [inline docker-compose-localstack.yaml:9]: Makes it look tidier in docker desktop

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiService.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+  fun getActivity(activityId: Long): Activity {
+    return webClient.get()
+      .uri("/activities/$activityId")
+      .retrieve()
+      .bodyToMono(Activity::class.java)
+      .block()!!
+  }
+}
+
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiService.kt:29]: Could maybe put these data classes in separate files

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiService.kt [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+// Copied from https://github.com/ministryofjustice/hmpps-activities-management-api
+data class ActivitySchedule(
+
+  @Schema(description = "The internally-generated ID for this activity schedule", example = "123456")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiService.kt:33]: not sure we need to `schema` annotations?

File: helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml [hunk 4]
```diff
@@ -51,6 +51,8 @@ generic-service:
       VISITS_CLIENT_SECRET: "VISITS_CLIENT_SECRET"
       INCENTIVES_CLIENT_ID: "INCENTIVES_CLIENT_ID"
       INCENTIVES_CLIENT_SECRET: "INCENTIVES_CLIENT_SECRET"
+      ACTIVITIES_CLIENT_ID: "ACTIVITIES_CLIENT_ID"
+      ACTIVITIES_CLIENT_SECRET: "ACTIVITIES_CLIENT_SECRET"
```
Comment (andymarke) [inline helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml:55]: Have these been created in live? Else this can't be merged since deployment will fail

**Review by andymarke — CHANGES_REQUESTED** [review]
Looks good but don't you have to add the activities url to the helm config else this will fail to start right?

**Review by andymarke — APPROVED** [review]
lgtm

File: helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml (hunk 4 above)
Comment (steverendell) [reply helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml:55]: I raised https://dsdmoj.atlassian.net/browse/HAAR-1092 to cover it. Ill do dev myself if Simon doesnt get round to it tomorrow

**Pull request merged**